flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryUtils.java

```diff
@@ -187,4 +188,10 @@ public static Table tableInfo(
                                 .setPrettyPrint(false)
                                 .execute());
     }
+
+    public static boolean tableExists(BigQueryConnectOptions connectOptions) {
+        BigQueryServices.QueryDataClient queryDataClient =
+                BigQueryServicesFactory.instance(connectOptions).queryClient();
+        return queryDataClient.tableExists(connectOptions.getDataset(), connectOptions.getTable());
```

Comment:
Ideally, we create new clients using `try with resource`: https://engdoc.corp.google.com/eng/doc/devguide/java/practices/closing.md?cl=head#try

Reply:
Also, there is a deviation in this method from the pattern followed by other classes in this file. All other methods similar to this one expect the client in their argument, instead of creating a new one.

I don't think the redirection through `BigQueryUtils` is needed. You have replicated the `tableExists` functionality in `BigQueryTableInfo` and `BigQueryUtils`. Implement the method in `BigQueryTableInfo` and expose it via `BigQueryService`

Reply:
We should anyway steer away from naming files as `Utils`: go/no-utils

Reply:
Couple more observations:
1. There is a deviation in this method from the pattern followed by others in this file. Other methods expect the client in the argument, while you are creating a new client in this method. These patterns are important as they reflect key principles like separation of responsibilities between different classes.
2. The tableExists method is present in BigQueryTableInfo and BigQueryUtils, which is repetitive. If you observe, these two classes have no overlapping methods. Please implement tableExists in BigQueryTableInfo, and expose via BigQueryServicesImpl. We want to steer away from naming files as `Utils`: [go/no-utils](https://docs.google.com/document/d/1MXVlsD4UlhOxgQPRMVLx88Uw9qLRqu8pEtpDuacyxXM)

Reply:
Thanks for pointing this out! Removed the redirection via BigQueryUtils.

Though now to invoke the function, 
`BigQueryServices.QueryDataClient queryDataClient = BigQueryServicesFactory.instance(connectOptions).queryClient();` has to be created.

This is similar to the invocation of `getTableSchemaFromOptions` at `BigQuerySchemaProviderImpl`

Should I add a redirection there?

Reply:
Yeah, create the client where you need to check existence of table, and invoke the `tableExists` method, similar to `getTableSchemaFromOptions`.

And, I just realized that the QueryDataClient is not auto-closeable. Basically, wrapping it in `try with resource` will make no difference. I think the instance will be cleared from memory when method's stack is gargabe collected eventually. Thats ok for now. We can revisit query client optimization later.

Reply:
Got it! But I believe making the call to check if the table exists, should be a part of another PR where the table creation is deferred. We can use this code snippet to invoke the function there.

Since `datasetExists()` PR #187  function also uses `QueryDataClient` we can reuse it as well.